internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -395,7 +396,11 @@ func (i *Infra) deleteServiceAccount(ctx context.Context, r ResourceRender) (err
 		}
 	}()
 
-	return i.Client.Delete(ctx, sa)
+	return i.Client.DeleteAllOf(ctx, sa, &client.DeleteAllOfOptions{
+		ListOptions: client.ListOptions{
+			LabelSelector: r.LabelSelector(),
```

Comment:
we should be using `stableSelector()` here 
https://github.com/envoyproxy/gateway/blob/1f29518dc2472d3f904d07eb93778675a4bb2ea3/internal/infrastructure/kubernetes/proxy/resource_provider.go#L205